screens/Menu/Settings.js

```diff
@@ -15,21 +32,128 @@ import PropTypes from "prop-types";
+                setAccessToken(result.accessToken);
+                return getUsersCalendarList(result.accessToken); //Here We are getting all the calendars in a json...
+            } else {
+                return { cancelled: true };
+            }
+        } catch (e) {
+            return { error: true };
+        }
+    }
+
+    const getFilteredGoogleCalendarList = (respCalendars) => {
+        var filteredList = respCalendars.items.map(element => {
+            return { id: element.id, summary: element.summary, description: element.description };
+        });
+        setCalendarList(filteredList);
+    };
+
+    const getUsersCalendarList = async (accessToken) => {
+        let calendarsList = await fetch('https://www.googleapis.com/calendar/v3/users/me/calendarList', {
+            headers: { Authorization: `Bearer ${accessToken}` },
+        });
+        let resp = await calendarsList.json();
+        return resp;
+    }
+
+    const press = (item) => {   // The onPress method 
+        var { checked } = isConnected;
+        // These ensures that multiple checkboxes don't all get affected when one is clicked
+        if (!checked.includes(item) || checked.length == 1) {
+            checked = [];
+            setIsConnected({ checked: [...checked, item] });
+        } else {
+            setIsConnected({ checked: checked.filter(a => a != item) });
+        }
+    };
+
+
+    // //CHECK IF IT WORKS
+    // const restrictSizeOfLabels = (filteredList) => {
+    //     var restrictedSizeList = filteredList.items.map(element => {
+    //         //If both the title and the description of the calendar is greater than 20 char, add ...
+    //         if (element.summary.length > 20 && element.description.length > 20) {
+    //             updatedSummary = element.summary.substring(0, 20) + "...";
+    //             updatedDescription = element.description.substring(0, 20) + "...";
+    //             return { id: element.id, summary: element.updatedSummary, description: element.updatedDescription };
+    //         }
+    //         //If the description of the calendar is greater than 20 char, add ...
+    //         if (element.description.length > 20) {
+    //             updatedDescription = element.description.substring(0, 20) + "...";
+    //             return { id: element.id, summary: element.summary, description: element.updatedDescription };
+    //         }
+    //         //If the title of the calendar is greater than 20 char, add ...
+    //         if (element.summary.length > 20) {
+    //             updatedDescription = element.summary.substring(0, 20) + "...";
+    //             return { id: element.id, summary: element.summary, description: element.updatedDescription };
+    //         }
+    //         else {
+    //             return { id: element.id, summary: element.summary, description: element.description };
+    //         }
+    //     });
+    //     setCalendarList(restrictedSizeList);
+    // }
```

Comment:
What are we doing with this commented code?

Reply:
I'm gonna to have wait on @badred123 final say so on that matter